dashboard/test/models/user_test.rb

```diff
@@ -1394,6 +1394,11 @@ def track_progress(user_id, script_level, result, pairings: nil)
     refute student.can_pair_with?(teacher)
     refute teacher.can_pair_with?(student)
     refute student.can_pair_with?(student)
+
+    # disable pair programming
+    section.update!(pairing_allowed: false)
+    student.reload
+    refute student.can_pair?
```

Comment:
This test is getting unreasonably long... Any chance this could be refactored into multiple tests as part of this PR?